tests/unittest/test_async_session.py

```diff
@@ -64,6 +64,13 @@ async def test_options(server):
         assert r.status_code == 200
 
 
+async def test_base_url(server):
+    async with AsyncSession(base_url=str(server.url)) as s:
+        r = await s.get("/echo_params", params={"foo": "bar"})
```

Comment:
What would be happening if pass absolute url at here again?

Reply:
If we pass an absolute URL when we have set `base_url`, it should use the absolute URL directly.
I'll add more tests to cover more cases.

Reply:
```py
async with AsyncSession(base_url="https://example.org") as s:
    r = await s.get("/echo_params", params={"foo": "bar"})
    r2 = await s.get("https://github.com", params={"foo": "bar"})
```
I mean, is `r2` send request to `github.com` or `example.org`?
(I guess `httpx` will send it to `github.com`)

Reply:
Same as `httpx`, `r2` will send the request to `github.com` instead of `example.org`, because it's an absolute url.
```python
In [2]: async with requests.AsyncSession(base_url="https://example.org") as s:
   ...:
   ...:     r = await s.get("/echo_params", params={"foo": "bar"})
   ...:     r2 = await s.get("https://github.com", params={"foo": "bar"})
   ...:     print(r.url)
   ...:     print(r2.url)
https://example.org/echo_params?foo=bar
https://github.com/?foo=bar
```